diff --git a/activity.py b/activity.py
index 693ceba..aaa6187 100644
--- a/activity.py
+++ b/activity.py
@@ -6,11 +6,9 @@ gi.require_version('Gdk', '3.0')
 from gi.repository import Gtk, Gdk
 import pygame
 
-from sugar3.activity.activity import Activity
 from sugar3.graphics.toolbarbox import ToolbarBox
 from sugar3.graphics.style import GRID_CELL_SIZE
 from sugar3.activity.widgets import StopButton
-from sugar3.activity.widgets import ActivityToolbarButton
 from sugar3.graphics.radiotoolbutton import RadioToolButton
 
 import sugargame.canvas
@@ -19,11 +17,13 @@ import falabracman
 
 sys.path.append('..')  # Import sugargame package from top directory.
 
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import ExtendedActivityToolbarButton
 
-class Falabracman(Activity):
+class Falabracman(SugarCompatibleActivity):
 
     def __init__(self, handle):
-        Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         self.paused = False
 
@@ -53,7 +53,7 @@ class Falabracman(Activity):
         self.set_toolbar_box(toolbar_box)
         toolbar_box.show()
 
-        activity_button = ActivityToolbarButton(self)
+        activity_button = ExtendedActivityToolbarButton(self)
         toolbar_box.toolbar.insert(activity_button, -1)
         activity_button.show()
 
